backend/core/taps/utils.py

```diff
@@ -114,83 +126,89 @@ def mformat_validator(msg):
 
     optional_fields_init = {"communities": []}
 
-    def valid_dict(msg):
-        if not isinstance(msg, dict):
+    def validate(self, msg):
+        self.msg = msg
+        if not self.valid_dict():
             return False
+
+        self.add_optional_fields()
+
+        for func in self.valid_generator():
+            if not func():
+                return False
+
         return True
 
-    def add_optional_fields(msg):
-        for field in optional_fields_init:
-            if field not in msg:
-                msg[field] = optional_fields_init[field]
+    def valid_dict(self):
+        if not isinstance(self.msg, dict):
+            return False
+        return True
 
-    def valid_fields(msg):
-        if any(field not in msg for field in mformat_fields):
+    def add_optional_fields(self):
+        for field in self.optional_fields_init:
+            if field not in self.msg:
+                self.msg[field] = self.optional_fields_init[field]
+
+    def valid_fields(self):
+        if any(field not in self.msg for field in self.mformat_fields):
             return False
         return True
 
-    def valid_prefix(msg):
+    def valid_prefix(self):
         try:
-            str2ip(msg["prefix"])
+            str2ip(self.msg["prefix"])
         except BaseException:
             return False
         return True
 
-    def valid_service(msg):
-        if not isinstance(msg["service"], str):
+    def valid_service(self):
+        if not isinstance(self.msg["service"], str):
             return False
         return True
 
-    def valid_type(msg):
-        if msg["type"] not in type_values:
+    def valid_type(self):
+        if self.msg["type"] not in self.type_values:
             return False
         return True
 
-    def valid_path(msg):
-        if msg["type"] == "A" and not isinstance(msg["path"], list):
+    def valid_path(self):
+        if self.msg["type"] == "A" and not isinstance(self.msg["path"], list):
             return False
         return True
 
-    def valid_communities(msg):
-        if not isinstance(msg["communities"], list):
+    def valid_communities(self):
+        if not isinstance(self.msg["communities"], list):
             return False
-        for comm in msg["communities"]:
+        for comm in self.msg["communities"]:
             if not isinstance(comm, dict):
                 return False
-            if community_keys - set(comm.keys()):
+            if self.community_keys - set(comm.keys()):
                 return False
         return True
 
-    def valid_timestamp(msg):
-        if not isinstance(msg["timestamp"], float):
+    def valid_timestamp(self):
+        if not isinstance(self.msg["timestamp"], float):
             return False
-        return True
-
-    def valid_peer_asn(msg):
-        if not isinstance(msg["peer_asn"], int):
+        if HISTORIC == "false" and datetime.utcfromtimestamp(
```

Comment:
here is the new addition where we check if the timestamp is smaller that the 1.5 hours before from now